src/Compilers/CSharp/Portable/Declarations/SingleTypeDeclaration.cs

```diff
@@ -245,7 +245,7 @@ public bool Equals(TypeDeclarationIdentity other)
                     return false;
                 }
 
-                if ((object)thisDecl.Location.SourceTree != otherDecl.Location.SourceTree
+                if ((object)thisDecl.SyntaxReference.SyntaxTree != otherDecl.SyntaxReference.SyntaxTree
```

Comment:
the Location was created from teh SyntaxReference, so this just avoids the unnecessary middle alloc.